File: exams/1617/translate/argparse/solution/argparse/flag-parameter.cpp
#include "flag-parameter.h"


FlagParameter::FlagParameter(const std::string& name)
	: Parameter(name), m_set(false) { }

bool FlagParameter::parse(std::list<std::string>& args)
{
	if (!args.empty() && args.front() == "--" + m_name)
	{
		args.pop_front();
		m_set = true;

		return true;
	}
	else
	{
		return false;
	}
}

bool FlagParameter::is_set() const
{
	return m_set;
}

std::shared_ptr<FlagParameter> flag(const std::string& name)
{
	return std::make_shared<FlagParameter>(name);
}

File: exams/1617/translate/argparse/student/java/FlagParameter.java
import java.util.ArrayList;


class FlagParameter extends Parameter
{
    private boolean set;

    public FlagParameter(String name)
    {
        super(name);
    }

    public boolean isSet()
    {
        return set;
    }

    public boolean parse(ArrayList<String> args)
    {
        if ( !args.isEmpty() && args.get(0).equals("--" + name) )
        {
            args.remove(0);
            set = true;

            return true;
        }
        else
        {
            return false;
        }
    }

    // Te vertalen als globale functie, d.i. een functie buiten de klasse
    public static FlagParameter flag(String name)
    {
        return new FlagParameter(name);
    }
}
